core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -229,7 +230,15 @@ public static void toJson(MetadataUpdate metadataUpdate, JsonGenerator generator
         writeAddSnapshot((MetadataUpdate.AddSnapshot) metadataUpdate, generator);
         break;
       case REMOVE_SNAPSHOTS:
-        writeRemoveSnapshots((MetadataUpdate.RemoveSnapshot) metadataUpdate, generator);
+        // TODO: Remove condition once RemoveSnapshot is deprecated and removed
```

Comment:
I think we can remove the TODO here and further above, because once `RemoveSnapshot` is actually removed, the code won't compile and this code will need to be updated anyway